### DISCUSSION TITLE
Erro ao criar migrations

### DISCUSSION THREAD

**Discussion by RafaelDifini** [balta-io/forum#112] [Tira Dúvidas] [answered]
CURSO: [Fundamentos do Azure, Git, GitHub e DevOps (89%)](https://balta.io/cursos/fundamentos-azure-git-github-devops)

seguindo o curso, chegando na parte de criar novamente as migrations da API, estou tendo o seguinte erro apos igualar as adequaçoes feitas no curso.
![image](https://user-images.githubusercontent.com/79666617/202581238-6423a74b-d9cd-4b2e-b921-996094fd9583.png)

meu DataContext esta assim:
![image](https://user-images.githubusercontent.com/79666617/202581294-4949f169-7fbd-423a-8607-2f1fd7f07ba8.png)

não tive problemas nesta migration no curso de ASP.NET

até entao, minha connection string local esta assim
![image](https://user-images.githubusercontent.com/79666617/202581399-a3434125-bc52-4c18-89e9-081bdae56602.png)

**Comment by andrebaltieri** [comment, marked as answer]
@RafaelDifini como está configurado o DbContext no seu Program.cs?

**Reply by RafaelDifini** [reply]
![image](https://user-images.githubusercontent.com/79666617/202818069-87866255-727d-4547-91c2-59b8558e06fa.png)

**Reply by RafaelDifini** [reply]
![image](https://user-images.githubusercontent.com/79666617/202818183-1df3aeb9-4b34-4973-bc40-9d79f4479b26.png)

**Comment by andrebaltieri** [comment]
Show, muito obrigado @RafaelDifini!!!

Vamos por partes... primeiro tenta criar um construtor vazio no seu DbContext... Veja se funciona aí...
Se não rolar, tenta informar o caminho do dbContext assim:
```csharp
services.AddDbContext<ApplicationDbContext>(
    options =>
        options.UseSqlServer(
            Configuration.GetConnectionString("DefaultConnection"),
            x => x.MigrationsAssembly("WebApplication1.Migrations")));
```

**Reply by RafaelDifini** [reply]
Balta, a classe ja tinha um construtor vazio, porem persiste o mesmo problema, tentei voltar o codigo como estava no curso anterior e mesmo assim continua, tbm tentei deixar exatamente igual ao seu codigo do 2812 e nada.

**Reply by RafaelDifini** [reply]
segue meu repo, caso prefira olhar: https://github.com/RafaelDifini/Blog-API

realmente travei no curso por isso, só falta isso pra testar a api no azure, e daqui a pouco acaba meus creditos gratis kkk @andrebaltieri

**Reply by andrebaltieri** [reply]
Executa isto por favor:
```
dotnet clean
dotnet build
```

Veja se está compilando tudo certo... 

Em adicional, pela imagem que mandou na primeira mensagem, seu DbContext não está com um construtor sem parâmetros... imagino que você tenha adicionado para testes, correto?

Feito isto, ele vai conseguir criar uma instância do seu DbContext sem uma configuração... ai você pode fixar uma configuração assim:
```csharp
protected override void OnConfiguring(DbContextOptionsBuilder options)
    => options.UseSqlServer("SUA CONNECTION STRING");
```

**Reply by RafaelDifini** [reply]
@andrebaltieri , Tentei fazer da forma como voce me indicou, deu certo o deploy, porem a API nao funcionou localmente nem publicada, na hora de executar os metodos me retornava 500, entao testei da seguinte forma:
![image](https://user-images.githubusercontent.com/79666617/203662525-e1b59b78-8630-4818-88b8-2211db49489d.png)

![image](https://user-images.githubusercontent.com/79666617/203662370-61cac736-0fd2-4df3-bd77-49074296175b.png)

e funcionou localmente, conectada com o DB da azure inclusive, POREM, infelizte na hora do deploy, continua dando aquele erro nas actions, mesmo com a migration ja criada, por algum motivo que desconheço, quando utilizo do jeito que esta nos prints e no seu projeto, o dotnef ef database update não funciona

![image](https://user-images.githubusercontent.com/79666617/203662760-be6d5621-280e-4d18-9952-b0096ce766a9.png)

eu consigo fazer esse deploy se usar o dbContext com OnConfiguring, mas como eu disse, ai as requisiçoes param de funcionar

**Reply by RafaelDifini** [reply]
quando eu deixo DbCotext dessa forma:
![image](https://user-images.githubusercontent.com/79666617/203663828-63fd55c0-093a-4806-b08b-8eea1ea6ce88.png)


recebo esse erro:
![image](https://user-images.githubusercontent.com/79666617/203663807-e1111a1c-8421-485f-b768-cf68ee71197e.png)

**Reply by RafaelDifini** [reply]
RESOLVIDO DE VEZ AGORA BALTA! hahahaha, faltava isso aqui
![image](https://user-images.githubusercontent.com/79666617/203665201-514339b1-ea37-4294-8ede-0405d62bf4f5.png)
deixei da forma como vc indicou
